packages/less/src/less/parser/parser.js

```diff
@@ -444,6 +444,19 @@ const Parser = function Parser(context, imports, fileInfo) {
                         return;
                     }
 
+                    for (let i = 0; i < args.length; i++) {
+                        const arg = args[i];
+                        // Do not call function with variable argument if this file is being imported by another file
+                        // and said variable is being redefined by the importing file
+                        if (fileInfo.filename !== fileInfo.rootFilename
+                            && imports.contents[fileInfo.rootFilename].match(arg.name + '\\s*:')
```

Comment:
This doesn't feel right that this should be in the parser, or have anything to do with arguments. Isn't this an evaluation issue? Does this only happen with functions, or is this just an issue with lazy evaluation in general?

Reply:
Ah, it seems you're absolutely right. I got a bit ahead of myself because the problem in the linked issue was about functions, but upon closer inspection it seems to be a more general issue with variables in imports not being evaluated -- the same thing applies, for example, to operators. So yes, this would have been the wrong approach.

Reply:
Reducing the test-case down to the minimum number of failing lines makes it clearer why the crash happened: 
`@base-color: var(--primary-color); @dark-color: darken(@base-color, 50%);` which makes sense as the function was trying to process arguments that will be defined later.